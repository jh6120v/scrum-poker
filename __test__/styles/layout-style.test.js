import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import {
    Container, HeaderStyle, NextLink, PrevLink, Title, Wrapper
} from '../../src/styles/layout-style';

describe('test layout-style.', () => {
    it('test <Wrapper /> snapshot.', () => {
        const { container } = render(<Wrapper />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <Wrapper /> when render it.', () => {
        const { container } = render(<Wrapper />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('max-width', '400px');
        expect(tree).toHaveStyleRule('min-height', '100%');
        expect(tree).toHaveStyleRule('flex-wrap', 'wrap');
        expect(tree).toHaveStyleRule('flex-direction', 'column');
        expect(tree).toHaveStyleRule('margin', '0');
        expect(tree).toHaveStyleRule('margin', '0 auto', {
            media: '(min-width:480px)'
        });
    });

    it('test <HeaderStyle /> snapshot.', () => {
        const { container } = render(<HeaderStyle />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <HeaderStyle /> when render it.', () => {
        const { container } = render(<HeaderStyle />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('height', 'calc(44px + env(safe-area-inset-top))');
        expect(tree).toHaveStyleRule('position', 'fixed');
        expect(tree).toHaveStyleRule('left', '0');
        expect(tree).toHaveStyleRule('top', '0');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'space-between');
        expect(tree).toHaveStyleRule('color', '#fff');
        expect(tree).toHaveStyleRule('padding', 'env(safe-area-inset-top) 5px 0 5px');
        expect(tree).toHaveStyleRule('overflow', 'hidden');
        expect(tree).toHaveStyleRule('background-color', '#6cb8cf');
        expect(tree).toHaveStyleRule('z-index', '3');
        expect(tree).toHaveStyleRule('left', '50%', {
            media: '(min-width:480px)'
        });
        expect(tree).toHaveStyleRule('width', '400px', {
            media: '(min-width:480px)'
        });
        expect(tree).toHaveStyleRule('margin-left', '-200px', {
            media: '(min-width:480px)'
        });
    });

    it('test <Title /> snapshot.', () => {
        const { container } = render(<Title />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <Title /> when render it.', () => {
        const { container } = render(<Title />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('font-family', '\'Fredoka One\',Baloo');
        expect(tree).toHaveStyleRule('font-size', '20px');
        expect(tree).toHaveStyleRule('font-weight', 'bold');
    });

    it('test <PrevLink /> snapshot', () => {
        const { container } = render(<PrevLink />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <PrevLink /> when render it.', () => {
        const { container } = render(<PrevLink />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('min-width', '25px');
        expect(tree).toHaveStyleRule('height', '25px');
    });

    it('test <NextLink /> snapshot.', () => {
        const { container } = render(<NextLink />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <NextLink /> when render it.', () => {
        const { container } = render(<NextLink />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('min-width', '25px');
        expect(tree).toHaveStyleRule('height', '25px');
    });

    it('test <Container /> snapshot.', () => {
        const { container } = render(<Container />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <Container /> when render it.', () => {
        const { container } = render(<Container />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('height', '100%');
        expect(tree).toHaveStyleRule('flex-grow', '1');
        expect(tree).toHaveStyleRule('flex-shrink', '1');
        expect(tree).toHaveStyleRule('flex-basis', 'auto');
    });
});