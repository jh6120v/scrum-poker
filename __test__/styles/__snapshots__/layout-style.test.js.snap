// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test layout-style. test <Container /> snapshot. 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
}

<div
  class="c0"
/>
`;

exports[`test layout-style. test <HeaderStyle /> snapshot. 1`] = `
.c0 {
  width: 100%;
  height: calc(44px + env(safe-area-inset-top));
  position: fixed;
  left: 0;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  padding: env(safe-area-inset-top) 5px 0 5px;
  overflow: hidden;
  background-color: #6cb8cf;
  z-index: 3;
}

@media (min-width:480px) {
  .c0 {
    left: 50%;
    width: 400px;
    margin-left: -200px;
  }
}

<header
  class="c0"
/>
`;

exports[`test layout-style. test <NextLink /> snapshot. 1`] = `
.c0 {
  min-width: 25px;
  height: 25px;
}

<div
  class="c0"
/>
`;

exports[`test layout-style. test <PrevLink /> snapshot 1`] = `
.c0 {
  min-width: 25px;
  height: 25px;
}

<div
  class="c0"
/>
`;

exports[`test layout-style. test <Title /> snapshot. 1`] = `
.c0 {
  font-family: 'Fredoka One',Baloo;
  font-size: 20px;
  font-weight: bold;
}

<div
  class="c0"
/>
`;

exports[`test layout-style. test <Wrapper /> snapshot. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 400px;
  min-height: 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
}

@media (min-width:480px) {
  .c0 {
    margin: 0 auto;
  }
}

<div
  class="c0"
/>
`;
